app/Traits/Controller/v1/HasSemesters.php

```diff
@@ -17,17 +18,23 @@ trait HasSemesters
     /**
      * Retrieve the specified user.
      *
-     * @param  string $semester_id
+     * @param  string  $semester_id
+     * @param  boolean $hideData
      * @return Semester
      */
-    protected function getSemester(string $semester_id=null)
+    protected function getSemester(string $semester_id=null, bool $hideData=false)
     {
-        if ($semester_id) {
-            if ($semester_id !== 'current') {
-                return Semester::getSemester($semester_id);
-            }
+        if ($semester_id && $semester_id !== 'current') {
+            $semester = Semester::getSemester($semester_id);
+        } else {
+            $semester = Semester::getThisSemester();
         }
 
-        return Semester::getThisSemester();
+        if ($hideData) {
+            return $semester->hideData();
```

Comment:
Here, if the `semester` is `null` as it could kindly be, it will throw an error ^^

So let the Trait as it was, revert all changes here and make your changes in your controller

Reply:
Like: 

```php
use HasSemesters {
  HasSemesters::getSemester as protected getSemesterFromTrait;
}

protected function getSemester(...) {
  $semester = $this->getSemesterFromTrait(...);
 // Do your math : is null ?

return $semester->hideData();
}